mne/preprocessing/eog.py

```diff
@@ -248,3 +248,104 @@ def create_eog_epochs(raw, ch_name=None, event_id=998, picks=None, tmin=-0.5,
                         picks=picks, baseline=baseline, preload=preload,
                         reject_by_annotation=reject_by_annotation)
     return eog_epochs
+
+
+@verbose
+def regress_eog(epochs, verbose=None):
+
+    """Regress eye channel data from eeg data
+
+      Gratton, Coles, Donchin (1983) EMCP - Eye movement correction procedure.
+
+      Parameters
+      ----------
+      data :
+       either
+       : instance of Epoch
+          The epoched data, in which condition average is subtracted first
+       or
+       : instance of Raw
+          The raw data  
+      
+      eeg_chans : pick_types array of channels
+
+      eog_chans : pick_types array of eye channels, preferably bipolar horizontal and vertical
+
+      
+      Returns
+      -------
+      regress_eog_epochs : instance of Epochs with eeg correct and eog retained
+      or
+      regress_eog_raw : instance of Raw with eeg correct and eog retained
+          
+
+      Raw Version Notes
+      -----
+      https://cbrnr.github.io/2017/10/20/removing-eog-regression/
+
+      Epoch Version Notes
+      -----
+      Correct EEG data for EOG artifacts with regression
+      -compute the ERP in each condition
+      -subtract ERP from each trial
+      -subtract baseline (mean over all epoch)
+      -predict eye channel remainder from eeg remainder
+      -use coefficients to subtract eog from eeg
+
+    """
+
+    event_names = ['A_error','B_error','C_error','D_error','E_error','F_error','G_error']
+    i = 0
+    for key, value in sorted(epochs.event_id.items(), key=lambda x: (x[1], x[0])):
+      event_names[i] = key
+      i += 1
+
+    #select the correct channels and data
+    eeg_chans = pick_types(epochs.info, eeg=True, eog=False)
+    eog_chans = pick_types(epochs.info, eeg=False, eog=True)
+    original_data = epochs._data
+
+    #subtract the average over trials from each trial
+    rem = {}
+    for event in event_names:
+      data = epochs[event]._data
+      avg = np.mean(epochs[event]._data,axis=0)
+      rem[event] = data-avg
```

Comment:
You should use a linter to fix pep8 violations like these.

```suggestion
      rem[event] = data - avg
```

Reply:
yes, have adopted that for my other PR, will update this one soon